contrib/rabbitmq/pkg/adapter/adapter.go

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbitmq
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/sbcd90/wabbit"
+	"github.com/sbcd90/wabbit/amqp"
```

Comment:
Why or you use your implementation? As I understand wabbit is used for a test with AMQP.

Reply:
Hello @Justin2997, thanks for your comment. `wabbit` has two components. `wabbit/amqp` & `wabbit/amqptest`. `wabbit/amqp` is a wrapper around the official library `streadway/amqp`.
`wabbit/amqptest` is used for mocking  amqp api s. Hence, `wabbit/amqp` is used in `adapter.go` & `wabbit/amqptest` used in `adapter_test.go`.

Now, `wabbit/amqp` & `wabbit/amqptest`  has a minor problem. They are not exposing timestamp. This enhancement I made in my fork & used it.
In case you want to fully remove wabbit, please let me know. I'll just remove the 2nd commit.

Reply:
Thanks a lot. I am surprised that the first time I see wabbit. I think of open source use that best is to use the standard `streadway/amqp`so in the future people can change the code, but I am not sure. I will let other contributor judge that.

Reply:
Hello @Justin2997, thanks for your comment. I also kept the commits separated for the same reason. Just wanted to show the approach of using `wabbit` over standard `streadway/amqp` & also its advantages in unit testing & disadvantages too.